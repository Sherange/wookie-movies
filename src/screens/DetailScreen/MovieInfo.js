import React from 'react';
import {StyleSheet, View, Text} from 'react-native';
import PropTypes from 'prop-types';
import CastList from './CastList';
import {primaryTextColor, secondryColor} from '../../constants/theme'

const MovieInfo = props => {
  const {year, duration, overview, cast, director} = props;
  return (
    <View style={styles.movieInfoContainer}>
      <Text style={styles.movieInfoTextStyle} numberOfLines={1}>
        {year} | {duration} | {director}
      </Text>

      <Text style={styles.movieInfoSubTitle}>Overview</Text>

      <Text style={styles.movieDescriptionText}>{overview}</Text>

      <Text style={styles.movieInfoSubTitle}>Cast</Text>

      <CastList castList={cast} />
    </View>
  );
};

const styles = StyleSheet.create({
  movieInfoContainer: {
    flex: 1,
    marginHorizontal: 20,
  },
  movieInfoSubTitle: {
    color: primaryTextColor,
    fontSize: 16,
    marginVertical: 8,
  },
  movieInfoTextStyle: {
    color: secondryColor,
    letterSpacing: 4,
    fontSize: 14,
    marginVertical: 8,
  },
  movieDescriptionText: {
    color: primaryTextColor,
    opacity: 0.7,
    fontSize: 16,
    lineHeight: 25,
    marginVertical: 8,
  },
});

MovieInfo.propTypes = {
  year: PropTypes.string.isRequired,
  duration: PropTypes.string.isRequired,
  director: PropTypes.oneOfType([
    PropTypes.string.isRequired,
    PropTypes.array.isRequired,
  ]),
  overview: PropTypes.string.isRequired,
  cast: PropTypes.array.isRequired,
};

export default MovieInfo;
