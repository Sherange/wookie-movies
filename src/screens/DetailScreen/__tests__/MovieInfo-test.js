import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfo from '../MovieInfo';

it('renders correctly', () => {
  renderer.create(
    <MovieInfo
      cast={['Heath Ledger', 'Aaron Eckhart']}
      overview={'overview here'}
      year={'2021'}
      duration={'2h 30min'}
      director={'Christian Bale'}
    />,
  );
});

test('MovieInfo snapshot', () => {
  const snapshot = renderer
    .create(
      <MovieInfo
        cast={['Heath Ledger', 'Aaron Eckhart']}
        overview={'overview here'}
        year={'2021'}
        duration={'2h 30min'}
        director={'Christian Bale'}
      />,
    )
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});
