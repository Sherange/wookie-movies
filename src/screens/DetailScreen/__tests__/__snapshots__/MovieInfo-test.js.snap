// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginHorizontal": 20,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#47CFFF",
        "fontSize": 14,
        "letterSpacing": 4,
        "marginVertical": 8,
      }
    }
  >
    2021
     | 
    2h 30min
     | 
    Christian Bale
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "marginVertical": 8,
      }
    }
  >
    Overview
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "lineHeight": 25,
        "marginVertical": 8,
        "opacity": 0.7,
      }
    }
  >
    overview here
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 16,
        "marginVertical": 8,
      }
    }
  >
    Cast
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <RCTScrollView
      horizontal={true}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 100,
              "justifyContent": "center",
              "marginRight": 8,
              "paddingVertical": 8,
              "width": 100,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#313D4D",
                "borderRadius": 25,
                "height": 50,
                "opacity": 0.7,
                "width": 50,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 25,
                  "flex": 1,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                borderRadius={25}
                onLoad={[Function]}
                style={
                  Object {
                    "bottom": 0,
                    "flex": 1,
                    "height": undefined,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": undefined,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
                transition={true}
                transitionDuration={360}
              />
              <View
                accessibilityElementsHidden={false}
                collapsable={false}
                importantForAccessibility="yes"
                nativeID="animatedComponent"
                pointerEvents="auto"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "center",
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image__placeholder"
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#ffffff",
                        "fontSize": 25,
                        "textAlign": "center",
                      }
                    }
                  >
                    HL
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
                testID="RNE__Image__children__container"
              />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "opacity": 0.7,
                "paddingVertical": 8,
              }
            }
          >
            Heath Ledger
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 100,
              "justifyContent": "center",
              "marginRight": 8,
              "paddingVertical": 8,
              "width": 100,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#313D4D",
                "borderRadius": 25,
                "height": 50,
                "opacity": 0.7,
                "width": 50,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <View
              accessibilityIgnoresInvertColors={true}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 25,
                  "flex": 1,
                  "overflow": "hidden",
                  "position": "relative",
                }
              }
            >
              <Image
                borderRadius={25}
                onLoad={[Function]}
                style={
                  Object {
                    "bottom": 0,
                    "flex": 1,
                    "height": undefined,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": undefined,
                  }
                }
                testID="RNE__Image"
                theme={
                  Object {
                    "colors": Object {
                      "black": "#242424",
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "grey": "rgba(0, 0, 0, 0.54)",
                          "primary": "#2196f3",
                          "searchBg": "#dcdce1",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "default": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "grey": "#7d7d7d",
                          "primary": "#007aff",
                          "searchBg": "#dcdce1",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                        "web": Object {
                          "error": "#ff190c",
                          "grey": "#393e42",
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                      "white": "#ffffff",
                    },
                  }
                }
                transition={true}
                transitionDuration={360}
              />
              <View
                accessibilityElementsHidden={false}
                collapsable={false}
                importantForAccessibility="yes"
                nativeID="animatedComponent"
                pointerEvents="auto"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "flex": 1,
                      "height": undefined,
                      "justifyContent": "center",
                      "width": undefined,
                    }
                  }
                  testID="RNE__Image__placeholder"
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#ffffff",
                        "fontSize": 25,
                        "textAlign": "center",
                      }
                    }
                  >
                    AE
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": undefined,
                    "width": undefined,
                  }
                }
                testID="RNE__Image__children__container"
              />
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "opacity": 0.7,
                "paddingVertical": 8,
              }
            }
          >
            Aaron Eckhart
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
